control/mttrie.go

```diff
@@ -145,81 +141,161 @@ func (mtt *mttNode) Add(mt *metricType) {
 // Fetch collects all children below a given namespace
 // and concatenates their metric types into a single slice
 func (mtt *mttNode) Fetch(ns []string) ([]*metricType, error) {
-	node, err := mtt.find(ns)
-	if err != nil {
-		return nil, err
-	}
-
-	var children []*mttNode
-	if node.mts != nil {
-		children = append(children, node)
-	}
-	if node.children != nil {
-		children = gatherChildren(children, node)
-	}
-
+	children := mtt.fetch(ns)
 	var mts []*metricType
 	for _, child := range children {
 		for _, mt := range child.mts {
 			mts = append(mts, mt)
 		}
 	}
-
+	if len(mts) == 0 {
+		return nil, errorMetricsNotFound("/" + strings.Join(ns, "/"))
+	}
 	return mts, nil
 }
 
-// Remove removes all children below a given namespace
+// Remove removes all descendants nodes below a given namespace
 func (mtt *mttNode) Remove(ns []string) error {
 	_, err := mtt.find(ns)
 	if err != nil {
 		return err
 	}
 
-	//remove node from parent
 	parent, err := mtt.find(ns[:len(ns)-1])
 	if err != nil {
 		return err
 	}
+	// remove node from parent
 	delete(parent.children, ns[len(ns)-1:][0])
 
 	return nil
 }
 
-// Get works like fetch, but only returns the MT at the given node
-// and does not gather the node's children.
-func (mtt *mttNode) Get(ns []string) ([]*metricType, error) {
-	node, err := mtt.find(ns)
+// GetMetric works like GetMetrics, but only returns the single MT in the requested version (or in the latest if ver < 1)
+// and does NOT gather the node's children.
+func (mtt *mttNode) GetMetric(ns []string, ver int) (*metricType, error) {
+	mts, err := mtt.GetMetrics(ns, ver)
 	if err != nil {
 		return nil, err
 	}
-	if node.mts == nil {
-		return nil, errorMetricNotFound("/" + strings.Join(ns, "/"))
+	// there is an expectation that only one metric should be fitted
+	if len(mts) > 1 {
+		return nil, fmt.Errorf("Incoming namespace `%s` is too ambiguous (version: %d)", "/"+strings.Join(ns, "/"), ver)
 	}
-	var mts []*metricType
-	for _, mt := range node.mts {
+	return mts[0], nil
```

Comment:
Because it has to have one element at least after `GetMetrics()`, so there is no need to check it once again in `GetMetric()`. If there is no element, an error will be returned in [line 179](https://github.com/intelsdi-x/snap/pull/1220/files#diff-b40077b597d83dcd21ab10660b31e40aR179).

Please, see [mttrie.go/GetMetrics line 207](https://github.com/intelsdi-x/snap/pull/1220/files#diff-b40077b597d83dcd21ab10660b31e40aR190)